pkg/client/client_test.go

```diff
@@ -1,56 +1,660 @@
+type Object interface {
+	metav1.Object
+	runtime.Object
+}
+
+type testInfo struct {
+	name    string
+	fields  clientFields
+	args    testArgs
+	wantErr bool
+}
+type clientFields struct {
+	Namespaced bool
+	GVR        schema.GroupVersionResource
+	Kind       string
+}
+type testArgs struct {
+	obj       Object
+	namespace string
+	desired   runtime.Object
+	result    runtime.Object
+}
+
+func TestClient_Get(t *testing.T) {
+	t.Parallel()
+	tests := []*testInfo{
+		{
+			name: "base test",
+			fields: clientFields{
+				GVR: schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				},
+				Namespaced: true,
+				Kind:       "Pod",
+			},
+			args: testArgs{
+				obj: &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+				},
+				namespace: "bar",
+				result:    &v1.Pod{},
+				desired: &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "foo",
+						Namespace:   "bar",
+						Annotations: map[string]string{"superhero": "batman"},
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Pod",
+						APIVersion: "v1",
+					},
+				},
+			},
+		},
+		{
+			name:    "empty name",
+			wantErr: true,
+			fields: clientFields{
+				GVR: schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				},
+				Namespaced: true,
+				Kind:       "Pod",
+			},
+			args: testArgs{
+				obj: &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "bar",
+					},
+				},
+				namespace: "bar",
+				result:    &v1.Pod{},
+				desired: &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "foo",
+						Namespace:   "bar",
+						Annotations: map[string]string{"superhero": "batman"},
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Pod",
+						APIVersion: "v1",
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		test := tt
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			mockRESTClient := &fake.RESTClient{
+				GroupVersion:         test.fields.GVR.GroupVersion(),
+				NegotiatedSerializer: negotiatedSerializer,
+			}
+			c := NewClient(test.fields.GVR, test.fields.Kind, test.fields.Namespaced, mockRESTClient, 0)
+			handler := newRequestHandler(c, test.args.desired, test.args.namespace, false, false)
+			mockRESTClient.Client = fake.CreateHTTPClient(handler)
+			err := c.Get(context.TODO(), test.args.namespace, test.args.obj.GetName(), test.args.result, metav1.GetOptions{})
+			if test.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, test.args.desired, test.args.result)
+		})
+	}
 }
 
-func TestWithAgent(t *testing.T) {
-	testUserAgent := "testagent"
-	testConfig := rest.Config{
-		UserAgent: "defaultuseragent",
-		ContentConfig: rest.ContentConfig{
-			NegotiatedSerializer: serializer.WithoutConversionCodecFactory{},
+func TestClient_List(t *testing.T) {
```

Comment:
Ignoring this since I didn't change the function. I just added some basic tests since there were none.

Reply:
Technically you did change the function since you changed the function signature of  `setupCtx` which is used by this function (requiring you to change this function in turn).

However, your point is taken about you not changing the actual behavior of the function, so I'll resolve the discussion.